lib/builtin.t

```diff
@@ -176,4 +176,79 @@ package FetchStoreCounter {
     cmp_ok($val, $_,  !1, "false is equivalent to  !1 by $_") for qw( eq == );
 }
 
+# Vanilla trim tests
+{
+    use builtin qw( trim );
+
+    is(trim("    Hello world!   ")      , "Hello world!"  , 'Trim spaces');
+    is(trim("\tHello world!\t")         , "Hello world!"  , 'Trim tabs');
+    is(trim("\n\n\nHello\nworld!\n")    , "Hello\nworld!" , 'Trim \n');
+    is(trim("\t\n\n\nHello world!\n \t"), "Hello world!"  , 'Trim all three');
+    is(trim("Perl")                     , "Perl"          , 'Trim nothing');
+    is(trim('')                         , ""              , 'Trim empty string');
+}
+
+TODO: {
+    my $warn = '';
+    local $SIG{__WARN__} = sub { $warn .= join "", @_; };
+
+    is(builtin::trim(undef), ""                                        , 'Trim undef');
+    like($warn    , qr/^Use of uninitialized value in subroutine entry at/, 'Trim undef triggers warning');
+    local $main::TODO = "Currently uses generic value for the name of non-opcode builtins";
+    like($warn    , qr/^Use of uninitialized value in trim at/,
+         'Trim undef triggers warning using actual name of builtin');
+}
+
+# Fancier trim tests against a regexp and unicode
+{
+    use builtin qw( trim );
+
+    is(trim("    Hello world!    "), trim_regexp_compare("    Hello world!    "), 'Trim compared to regexp');
+    is(trim("\n\nHello world!    "), trim_regexp_compare("\n\nHello world!    "), 'Trim compared to regexp');
+    is(trim("    Hello world!\t\t"), trim_regexp_compare("    Hello world!\t\t"), 'Trim compared to regexp');
+    is(trim("   \N{U+2603}       "), "\N{U+2603}"                               , 'Trim with unicode content');
+    is(trim("\N{U+2029}foobar    "), "foobar"                                   , 'Trim with unicode whitespace');
+    is(trim("\xa0foobar          "), "foobar"                                   , 'Trim with latin1 whitespace');
+}
+
+# Tests against special variable types and scopes
```

Comment:
The `my` case is the same as previous ones; the `our` can probably just live in the top block with those

Reply:
done